lib/puppet/provider/mongodb_database/mongodb.rb

```diff
@@ -6,19 +6,23 @@
 
   def self.instances
     require 'json'
-    dbs = JSON.parse mongo_eval('printjson(db.getMongo().getDBs())')
+    begin
+      dbs = JSON.parse mongo_eval('printjson(db.getMongo().getDBs())')
 
-    dbs['databases'].map do |db|
-      new(name: db['name'],
-          ensure: :present)
+      dbs['databases'].map do |db|
+        new(name: db['name'],
+            ensure: :present)
+      end
+    rescue
```

Comment:
Should we log anything about why it failed? Otherwise debugging might be very hard.